src/TweetProcessor.js

```diff
@@ -277,11 +277,21 @@ class TweetProcessor {
       promise = Promise.resolve();
     }
 
-    promise = promise.then(() => { return this._twitter.profile.update(tweet.user); });
+    // If all the rooms are marked as hot, don't bother with updating the profile.
+    const updateProfiles = ![...rooms].every((roomId) => {return opts.hotRooms.has(roomId)});
+    if (updateProfiles) {
+      promise = promise.then(() => { return this._twitter.profile.update(tweet.user); });
+    }
+    else {
+      promise = Promise.resolve();
```

Comment:
This is a bug because you might end up overwriting the promise created on line 268. I think this `else` is redundant.